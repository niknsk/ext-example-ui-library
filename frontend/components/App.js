// Copyright 1999-2017. Plesk International GmbH. All rights reserved.

import { createElement, Alert } from '@plesk/ui-library';

const App = ({ date }) => (
    <Alert intent="info">
        Server time: {date}
    </Alert>
);

export default App;
