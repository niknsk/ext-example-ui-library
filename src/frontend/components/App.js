// Copyright 1999-2017. Plesk International GmbH. All rights reserved.

import { createElement, Component, React } from '@plesk/ui-library';
import Header from './Header/Header';
import Main from './Main';

class App extends Component {
    render() {
        return (
            <div>
                <Header />
                <Main />
            </div>
        );
    }
}

export default App;
