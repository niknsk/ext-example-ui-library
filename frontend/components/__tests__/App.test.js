// Copyright 1999-2017. Plesk International GmbH. All rights reserved.

import { createElement } from '@plesk/ui-library';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <App date="2017-08-16 12:06:57" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
