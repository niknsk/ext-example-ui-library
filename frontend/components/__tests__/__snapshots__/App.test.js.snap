// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="pul-alert pul-alert--info"
>
  <span
    className="pul-icon"
  >
    <img
      alt=""
      src="images/info.png"
    />
  </span>
   
  Server time: 
  2017-08-16 12:06:57
</div>
`;
